src/lib.rs

```diff
@@ -2701,52 +2722,38 @@ impl<
     /// Get the status of a transaction.
     pub async fn transaction_status(
         &self,
-        receipt: &TransactionReceipt<L>,
+        uid: &TransactionUID<L>,
     ) -> Result<TransactionStatus, KeystoreError<L>> {
-        let KeystoreSharedState { state, model, .. } = &mut *self.mutex.lock().await;
-        state.transaction_status(model, receipt).await
+        let KeystoreSharedState {
+            state: _, model, ..
+        } = &mut *self.mutex.lock().await;
+        Ok(model.transactions.get(uid)?.status())
     }
 
     /// A future which completes when the transaction is finalized (committed or rejected).
+    /// Works only for transactions we submitted
     pub async fn await_transaction(
         &self,
-        receipt: &TransactionReceipt<L>,
+        uid: &TransactionUID<L>,
     ) -> Result<TransactionStatus, KeystoreError<L>> {
         let mut guard = self.mutex.lock().await;
         let KeystoreSharedState {
-            state,
+            state: _,
             model,
             txn_subscribers,
-            pending_foreign_txns,
             ..
         } = &mut *guard;
-
-        let status = state.transaction_status(model, receipt).await?;
+        let status = model.transactions.get(uid)?.status();
```

Comment:
@jbearer The cause of the deadlock was actually here not with `AtomicStore::commit_version` for the most recent test timeouts.  I had changed this to call `self.transaction_status()` which locks the shared state again